CHANGELOG.md

```diff
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Minimum Rust version has been bumped to 1.37.0
 
+### Added
+
+- Modal keyboard cursor motion mode for copying text
```

Comment:
... and opening URLs.

Reply:
Actually, just a little more thinking about this and I think I have a better idea.

The name "keyboard motion" has **really** been bothering me. For example, it's obvious it's a keyboard something when bound to a key in `key_bindings`. So a better name for the key, in place of `action` would be `cursor_motion`. However, that causes a small issue, which is that we want a good name for the mode too.

So why not just call the whole thing simply, "motion". Then we have:

```yml
- { key: J, motion: Down }
```
instead of:
```yml
- { key: J, keyboard_motion: Down }
```
which I contest, makes **a lot** more sense.

Additionally, we'd have:

```yml
- { key: G, mode: Motion, action: ScrollToTop }
```
Which, isn't necessarily any more sensible, but it's also no worse IMO, and it's shorter.

Reply:
> Keyboard motion mode for moving the cursor, which allows selecting text to copy and URLs to open

Maybe? Though `motion mode for moving the cursor` sounds similarly redundant. I'll do some more brainstorming.

Reply:
Either way we have a slight problem, `Toggle*Selection`s are **not** motions, yet they must be labeled as such. This is very surprising to me, and is sure to cause confusion. I'd expect all motions to actually move the cursor.

Why can't these be regular actions? For example, maybe I want to convert my normal selection into a block selection. This could be done without entering motion mode, right? I don't see any reason to ship these actions as defaults outside of motion mode, but I don't see why they aren't actions.

Reply:
> So a better name for the key, in place of action would be cursor_motion

I ***STRONGLY*** disagree with this suggestion. Neither keyboard nor cursor suggest at all what this mode does. Missing the connection to the keyboard for allowing motion with it will be extremely confusing. We already have a cursor and it can already move around. Cursor motion does not help at all with understanding what's going on.

The new concept is that there is a new cursor linked to the keyboard, which can be moved around with it. Which keyboard_motion and keyboard_motion_cursor express. It allows you to move around with the keyboard and it has a new cursor for it.

Reply:
> Why can't these be regular actions? For example, maybe I want to convert my normal selection into a block selection. This could be done without entering motion mode, right? I don't see any reason to ship these actions as defaults outside of motion mode, but I don't see why they aren't actions.

This has been discussed before and we've decided against it, since its behavior outside of keyboard motion mode felt strange and unnatural.

Reply:
> If a user wants a keybinding to open a URL while not in motion mode, I don't see the issue with that.

The keyboard motion cursor is not the same as the mouse cursor. There is no concept of "URL under the cursor" when not inside of the keyboard motion mode.

Reply:
> I STRONGLY disagree with this suggestion.

Sorry I should have been clear, I'm suggesting `motion` as the name.

Reply:
I think calling this vi mode is better. It'll invoke the correct keybindings for users, and is mostly consistent with what vi does.